### PR TITLE
Add methods to support editor links

### PR DESCRIPTION
**Pull request by markstory**
This adds methods to Debugger that will help generate editor links in error pages. I'll be updating the error pages in a separate pull request.

### Questions

* Are there other editors that we should include?
* Suggestions on where the editor links should be added in the error page?

### CODE REVIEW

File: src/Error/Debugger.php [hunk 1]
```diff
@@ -61,6 +62,7 @@ class Debugger
     protected $_defaultConfig = [
         'outputMask' => [],
         'exportFormatter' => null,
+        'editor' => 'phpstorm',
```
Comment (ADmad) [inline src/Error/Debugger.php]: This default should be configurable. The constructor could use `Configure::read('Debugger.editor')`.

Comment (markstory) [reply src/Error/Debugger.php]: :+1: on making it configurable from the application config. I was considering adding an `editor` option to the `Error` key, but if have a few other options having Debugger be separate could be nice.

Comment (ADmad) [reply src/Error/Debugger.php]: A key under `Error` works too.

**Comment by josegonzalez** [comment]
Maybe a few good ones to add

- textmate
- atom
- intellij

**Comment by markstory** [comment]
I've added atom and textmate. Intellij [seems more complicated though](https://youtrack.jetbrains.com/issue/IDEA-65879).

**Pull request merged**